nautobot/extras/models/groups.py

```diff
@@ -17,16 +18,153 @@
 from nautobot.core.models import BaseManager, BaseModel
-from nautobot.core.models.generics import OrganizationalModel
+from nautobot.core.models.generics import OrganizationalModel, PrimaryModel
 from nautobot.core.utils.config import get_settings_or_config
 from nautobot.core.utils.lookup import get_filterset_for_model, get_form_for_model
 from nautobot.extras.choices import DynamicGroupOperatorChoices
 from nautobot.extras.querysets import DynamicGroupMembershipQuerySet, DynamicGroupQuerySet
-from nautobot.extras.utils import extras_features
+from nautobot.extras.utils import extras_features, FeatureQuery
 
 logger = logging.getLogger(__name__)
 
 
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "webhooks",
+)
+class StaticGroup(PrimaryModel):
+    """A statically defined (as opposed to dynamically calculated) group of objects sharing a common content-type."""
+
+    name = models.CharField(max_length=CHARFIELD_MAX_LENGTH, unique=True)
+    description = models.CharField(max_length=CHARFIELD_MAX_LENGTH, blank=True)
+    content_type = models.ForeignKey(
+        to=ContentType,
+        on_delete=models.CASCADE,
+        related_name="static_groups",
+        limit_choices_to=FeatureQuery("static_groups"),
+        help_text="The type of object contained in this Static Group.",
+    )
+
+    clone_fields = ["content_type"]
+    is_static_group_associable_model = False
+
+    class Meta:
+        ordering = ["name"]
+
+    def __str__(self):
+        return self.name
+
+    @property
+    def model(self):
+        """
+        Access to the underlying Model class for this group's `content_type`.
+
+        This class object is cached on the instance after the first time it is accessed.
+        """
+        if getattr(self, "_model", None) is None:
+            try:
+                self._model = self.content_type.model_class()
+            except models.ObjectDoesNotExist:
+                self._model = None
+        return self._model
+
+    @property
+    def members(self):
+        """Return the member objects for this group."""
+        # Since associated_object is a GenericForeignKey, we can't just do:
+        #     return self.static_group_associations.values_list("associated_object", flat=True)
+        return self.model.objects.filter(
+            pk__in=self.static_group_associations.values_list("associated_object_id", flat=True)
+        )
+
+    def clean(self):
+        super().clean()
+
+        if self.present_in_database:
+            # Check immutable fields
+            database_object = self.__class__.objects.get(pk=self.pk)
+
+            if self.content_type != database_object.content_type:
+                raise ValidationError({"content_type": "ContentType cannot be changed once created"})
+
+    def _check_valid_member(self, member):
+        if not (isinstance(member, self.model) or issubclass(member, self.model)):
```

Comment:
Is `issubclass` valid? Wouldn't that allow more than one ContentType per static group?

Reply:
The intent here was to allow either model instances (list of objects case) or a specific model class (queryset case) to be validated. I hadn't thought about the concrete subclass case though, so good catch. I think you're right in your next comment that just checking equality should be sufficient for that case. Will update.